compiler/rustc_codegen_cranelift/src/abi/mod.rs

```diff
@@ -39,7 +39,7 @@ fn clif_sig_from_fn_abi<'tcx>(
 pub(crate) fn conv_to_call_conv(sess: &Session, c: Conv, default_call_conv: CallConv) -> CallConv {
     match c {
         Conv::Rust | Conv::C => default_call_conv,
-        Conv::RustCold => CallConv::Cold,
+        Conv::Cold | Conv::PreserveMost | Conv::PreserveAll => CallConv::Cold,
```

Comment:
None of these are ABI compatible between LLVM and Cranelift actually. So using them in the standard library will break ABI compatibility. It should be possible to implement in Cranelift if LLVM guarantees a specific call conv for them, but that doesn't help for GCC.

Reply:
Hmm, I guess that's not a blocker for this PR, though, since it was `coldcc` before, which if I understand correctly already wasn't ABI compatible.

What's your meta expectation here for the future?  Do all `extern`s have to be interoperable between backends?  That would seem a shame, as it would essentially force `extern "Rust"` to always be the same as `extern "C"`, rather than letting a backend innovate something faster.  Would it be ok to have options to force `rust-cold` to be something else (maybe just `C`) for people who need the interop?

Reply:
> Do all externs have to be interoperable between backends?

Preferably, but if any is not interoperable, denying usage in the standard library and emitting a hard error if caller and callee are compiled with a different codegen backend would work too I guess. (eg by internally lowering it to a different call conv depending on the selected codegen backend before we do typeck)

> That would seem a shame, as it would essentially force extern "Rust" to always be the same as extern "C", rather than letting a backend innovate something faster.

`extern "Rust"` could be changed to internally use eg `extern "sysv"` on all targets or `extern "win64"` on all x86_64 targets. Note that `extern "Rust"` is already not `extern "C"`. The lowering from rust types to primitives is different, but those primitives are passed in the same way such that it is actually possible for all backends to use the same ABI by reusing the lowering code in rustc_target. I don't want to change that.

> Would it be ok to have options to force rust-cold to be something else (maybe just C) for people who need the interop?

That wouldn't help if the standard library uses it as the LLVM compiled standard library will be used with cg_clif once it ships on nightly.